### PR TITLE
Upgrade guidance on upgrading Riak

### PR DESCRIPTION
**Pull request by kuenishi**
Regarding the result of RCS-147.

### CODE REVIEW

**Comment by kuenishi** [comment]
Some of my mistake on tests insights revealed. Will run test again.

**Comment by kuenishi** [comment]
Re-running the test, but these sentences are worth mentioning regardless of the test result. This is ready for review again.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -152,6 +152,30 @@ Including CS path: add following sentence to `advanced.config`:
 ]}.
 ```
 
+### Notes on upgrading Riak to 2.0
+
+Riak has updated its underlying Bitcask storage data format at 2.0. At
+the first start of Riak after upgrade, it involves implicit data
+format upgrade conversion, which means reading all data and writing
+down to another file. This might lead to disk load - the duration of
```
Comment (shino) [inline RELEASE-NOTES.md]: other files

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -152,6 +152,30 @@ Including CS path: add following sentence to `advanced.config`:
...
+upgrade will depend on the amount of data stored in bitcask and IO
+performance of underlying disk.
```
Comment (shino) [inline RELEASE-NOTES.md:163]: This paragraph explains only downside. I want one addition sentence for why format change was needed.

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -152,6 +152,30 @@ Including CS path: add following sentence to `advanced.config`:
...
+2015-03-17 07:18:49.754 [info] <0.609.0>@riak_kv_bitcask_backend:callback:446 Finished upgrading to Bitcask 1.7.0 in /mnt/data/bitcask/1096126227998177188652763624537212264741949407232
+2015-03-17 07:23:07.181 [info] <0.610.0>@riak_kv_bitcask_backend:callback:446 Finished upgrading to Bitcask 1.7.0 in /mnt/data/bitcask/1278813932664540053428224228626747642198940975104
+```
+
+while the start time is like this:
+
+```
+2015-03-17 02:43:20.813 [info] <0.609.0>@riak_kv_bitcask_backend:maybe_start_upgrade_if_bitcask_files:720 Starting upgrade to version 1.7.0 in /mnt/data/bitcask/1096126227998177188652763624537212264741949407232
+2015-03-17 02:43:21.344 [info] <0.610.0>@riak_kv_bitcask_backend:maybe_start_upgrade_if_bitcask_files:720 Starting upgrade to version 1.7.0 in /mnt/data/bitcask/1278813932664540053428224228626747642198940975104
+```
```
Comment (shino) [inline RELEASE-NOTES.md:178]: I think these log example are very helpful for user/operators :smile:

Small details, I prefer temporal order to present these logs.

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -233,8 +257,8 @@ following instructions for each nodes:
 Bitcask file format has changed between Riak 1.4 and 2.0. It supports
 implicit upgrade of bitcask data files, while downgrading is not
 supported. This is why downgrading requires a script to translate data
-files.
```
Comment (shino) [inline RELEASE-NOTES.md:236]: Not in this PR.

> while downgrading is not supported

`s/downgrading/automatic downgrading/`

**Comment by shino** [comment]
Added a few trivial comments. I will :+1: after fixing them.

**Comment by kuenishi** [comment]
Updated ^^;

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -232,9 +257,9 @@ following instructions for each nodes:
 
 Bitcask file format has changed between Riak 1.4 and 2.0. It supports
 implicit upgrade of bitcask data files, while downgrading is not
-supported. This is why downgrading requires a script to translate data
-files.
-
+supported. This is why automatic downgrading requires a script to
```
Comment (shino) [inline RELEASE-NOTES.md]: Sorry that my comment was unclear.

The sentence I wanted to point was: 

> Bitcask file format has changed between Riak 1.4 and 2.0. It supports
> implicit upgrade of bitcask data files, while downgrading is not supported.

At first glance, readers may think Riak 2.0 does not support downgrading to 1.4 (once bitcask was upgraded). I want to add "automatic" to the sentence like this:

> Bitcask file format has changed between Riak 1.4 and 2.0. It supports
> implicit upgrade of bitcask data files, while **automatic** downgrading is not supported.

Comment (kuenishi) [inline RELEASE-NOTES.md]: Ah, makes sense.

**Comment by kuenishi** [comment]
Updated @shino 👀

**Comment by kuenishi** [comment]
@borshop merge

**Pull request merged**